logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -88,6 +101,93 @@ public void moveValidSnapshot(long lowerBoundSnapshotPosition) throws IOExceptio
     Files.move(previousLocation.toPath(), snapshotDir.toPath());
   }
 
+  public void replicateLatestSnapshot(Consumer<Runnable> executor) {
+    final List<File> snapshots = storage.listByPositionDesc();
+
+    if (snapshots != null && !snapshots.isEmpty()) {
+      final File latestSnapshotDirectory = snapshots.get(0);
+      LOG.debug("Start replicating latest snapshot {}", latestSnapshotDirectory.toPath());
+      final long snapshotPosition = Long.parseLong(latestSnapshotDirectory.getName());
+
+      final File[] files = latestSnapshotDirectory.listFiles();
+      for (File snapshotChunk : files) {
+        LOG.debug("Replicate snapshot chunk {}", snapshotChunk.toPath());
```

Comment:
I think you should move the log statement inside the executor block :sweat_smile: